### PR TITLE
search functionality in app marketplace and bpn adress ui fixed

### PR DESCRIPTION
**Pull request by nidhigarg-bmw**

### CODE REVIEW

File: cx-portal/src/components/pages/AppMarketplace/components/AppListSection/index.tsx [hunk 1]
```diff
@@ -65,6 +76,33 @@ export default function AppListSection() {
...
+              (card) =>
+                card.title.toLowerCase().includes(expr.toLowerCase()) ||
+                (card.subtitle &&
+                  card.subtitle.toLowerCase().includes(expr.toLowerCase()))
+            )
+          )
+        } else {
+          setCardsData(cards)
+        }
+      }, 300),
```
Comment (oyo) [inline cx-portal/src/components/pages/AppMarketplace/components/AppListSection/index.tsx:94]: how about this more compact implementation?
`
debounce((expr: string) => setCardsData(
  expr
    ? cards.filter(...etc...)
    : cards
), 300)
`

**Review by oyo — CHANGES_REQUESTED** [review]
one small remark

File: cx-portal/src/components/pages/AppMarketplace/components/AppListSection/index.tsx (hunk 1 above)
Comment (nidhigarg-bmw) [reply cx-portal/src/components/pages/AppMarketplace/components/AppListSection/index.tsx]: @oyo  Done this change, please review again

**Review by oyo — APPROVED** [review]
lgtm now

**Pull request merged**